statsmodels/stats/multivariate.py

```diff
@@ -59,6 +59,41 @@ def test_mvmean(data, mean_null=0, return_results=True):
     else:
         return statistic, pvalue
 
+def test_mvmean_two_sample(x, y, return_results=True):
+    """Hotellings test for multivariate mean in two samples
+
+    Parameters
+    ----------
+    x : array_like
+        first sample data with observations in rows and variables in columns
+    y : array_like
+        second sample data with observations in rows and variables in columns
+
+    Returns
+    -------
+    results : instance of a results class with attributes
+        statistic, pvalue, t2 and df
+    """
+    x = array_like(x, "x", ndim=2)
+    y = array_like(y, "x", ndim=2)
+    nobs_x, k_vars = x.shape
+    nobs_y, k_vars = y.shape
+    mean_x = x.mean(0)
+    mean_y = y.mean(0)
+    cov_x = np.cov(x, rowvar=False, ddof=1)
+    cov_y = np.cov(y, rowvar=False, ddof=1)
+    combined_cov = ((nobs_x - 1) * cov_x + (nobs_y - 1) * cov_y) / (nobs_x - 1 + nobs_y - 1)
```

Comment:
you could use a temp variable `nobs_t = nobs_x + nobs_y`
I use nobs1, nobs2 in two sample case, also mean1, mean2, ...